osf/utils/datetime_aware_jsonfield.py

```diff
@@ -84,11 +84,14 @@ def formfield(self, **kwargs):
         defaults.update(kwargs)
         return super(DateTimeAwareJSONField, self).formfield(**defaults)
 
-    def from_db_value(self, value, expression, connection, context):
+    def from_db_value(self, value, expression, connection):
         if value is None:
-            return None
-        return super(DateTimeAwareJSONField, self).to_python(decode_datetime_objects(value))
+            return value
 
+        try:
+            return decode_datetime_objects(json.loads(value, cls=self.decoder))
+        except json.JSONDecodeError:
+            return value
```

Comment:
This might do

```python
# Customize `from_db_value()` to decode datetime objects
# Doc: https://docs.djangoproject.com/en/3.2/ref/models/fields/#jsonfield
def from_db_value(self, value, expression, connection):
    value = super(DateTimeAwareJSONField, self).from_db_value(value)
    return decode_datetime_objects(value)
```